core/domain/email_manager_test.py

```diff
@@ -2003,6 +2003,336 @@ def test_that_correct_completion_email_is_sent(self) -> None:
                 feconf.EMAIL_INTENT_REVIEW_CREATOR_DASHBOARD_SUGGESTIONS)
 
 
+class NotifyContributionAchievementEmailTests(test_utils.EmailTestBase):
+    """Test that correct email is sent while notifying contributor
+    achievements."""
+
+    USERNAME: Final = 'user'
+    USER_EMAIL: Final = 'user@example.com'
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.USER_EMAIL, self.USERNAME)
+        self.user_id = self.get_user_id_from_email(self.USER_EMAIL)  # type: ignore[no-untyped-call]
+        user_services.update_email_preferences(
+            self.user_id, True, False, False, False)
+        self.can_send_emails_ctx = self.swap(feconf, 'CAN_SEND_EMAILS', True)
+        self.can_not_send_emails_ctx = self.swap(
+            feconf, 'CAN_SEND_EMAILS', False)
+
+    def test_that_email_not_sent_if_can_send_emails_is_false(self) -> None:
+        contributor_ranking_email_info = (
+            suggestion_registry.ContributorMilestoneEmailInfo(
+                self.user_id, 'question', 'edit', None,
+                'Initial Contributor'
+            ))
+        with self.can_not_send_emails_ctx:
+            email_manager.send_mail_to_notify_contributor_ranking_achievement(
+                contributor_ranking_email_info)
+
+        messages = self._get_sent_email_messages(
+            self.USER_EMAIL)
+        self.assertEqual(len(messages), 0)
```

Comment:
```suggestion
        messages = self._get_sent_email_messages(
            self.USER_EMAIL)

        self.assertEqual(len(messages), 0)
```

Insert new line between action and assert to make test structure clear. Here and below.